src/gui/painting/qpainter.cpp

```diff
@@ -2673,6 +2673,158 @@ QPainterPath QPainter::clipPath() const
+{
+    return PathClipReplaceClip(matrix, info);
+}
+
+static QPainterPath RectClipReplaceClip(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    QPainterPath tempPath;
+    tempPath.addRect(info.rect);
+    tempPath = tempPath * matrix;
+    return tempPath;
+}
+
+static QPainterPath RectClipIntersectClip(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    return RectClipReplaceClip(matrix, info);
+}
+
+static QPainterPath RectFClipReplaceClip(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    QPainterPath tempPath;
+    tempPath.addRect(info.rectf);
+    return tempPath;
+}
+
+static QPainterPath RectFClipIntersectClip(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    // if you simplify this method, you can simplify clipped_path
+    // Use rect intersection if possible.
+    QPainterPath tempPath;
+
+    if (matrix.type() <= QTransform::TxScale) {
+        QPainterPath tempPath;
+        tempPath.addRect(matrix.mapRect(info.rectf));
+    } else {
+        QPainterPath tempPath;
+        tempPath.addRegion(matrix.map(QRegion(info.rectf.toRect())));
+    }
+    return tempPath;
+}
+
+static QPainterPath RegionClipReplaceClip(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    QPainterPath tempPath;
+    tempPath.addRegion(info.region * matrix);
+    return tempPath;
+}
+
+static QPainterPath RegionClipIntersectClip(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    return RegionClipReplaceClip(matrix, info);
+}
+
+/*!
+    Returns the current clip path in logical coordinates, unrounded.
+
+    The main purpose for this function is inside QSvgGenerator,
+    where we do not want any points to be rounded during the
+    creation of the file.
+
+    \warning QPainter does not store the combined clip explicitly as
+    this is handled by the underlying QPaintEngine, so the path is
+    recreated on demandand transformed to the current logical
+    coordinate system.This is potentially an expensive operation.
+
+    \sa setClipPath(), clipRegion(), setClipping()
+*/
+QPainterPath QPainter::clipPathF() const
+{
+    Q_D(const QPainter);
+
+    if (!d->engine) {
+        qWarning("QPainter::clipPath: Painter not active");
+        return QPainterPath();
+    }
+
+    // No clip, return empty
+    if (d->state->clipInfo.isEmpty()) {
+        return QPainterPath();
+    } else {
+
+        // Update inverse matrix, used below.
+        if (!d->txinv)
+            const_cast<QPainter *>(this)->d_ptr->updateInvMatrix();
+
+        /*
+        * Modified version of "clipRegion" that is used in ::clipPath
+        * as "return qt_regionToPath(clipRegion());" that prevents rounding of values,
+        * placing them directly into a QPainterPath.
+        */
+        QPainterPath path;
+        bool initializing = true;
+        if (!d->txinv)
+            const_cast<QPainter *>(this)->d_ptr->updateInvMatrix();
+
+
+        enum ActionForClipOperation {
```

Comment:
move, `ActionForClipOperation `, `PainterFunc` and `clipped_path` outside of function. there is no reason it should be performed dynamically when function is called.